beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3807,15 +3863,50 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         );
         self.import_block_update_slasher(block, &state, &mut consensus_context);
 
-        let db_write_timer = metrics::start_timer(&metrics::BLOCK_PROCESSING_DB_WRITE);
-
         // Store the block and its state, and execute the confirmation batch for the intermediate
         // states, which will delete their temporary flags.
         // If the write fails, revert fork choice to the version from disk, else we can
         // end up with blocks in fork choice that are missing from disk.
         // See https://github.com/sigp/lighthouse/issues/2028
         let (_, signed_block, blobs, data_columns) = signed_block.deconstruct();
+        let custody_columns_count = self.data_availability_checker.get_custody_columns_count();
+        let data_columns = data_columns.filter(|columns| columns.len() >= custody_columns_count);
```

Comment:
Why is this filter necessary? Why would we have too many columns? If that is the case why are the first columns in the vec more important than the latter?